jupyter-lib/shared-compiler/src/main/kotlin/org/jetbrains/kotlinx/jupyter/debug/DebugIntegration.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.kotlinx.jupyter.debug
+
+import kotlin.concurrent.thread
+
+abstract class DebugIntegrationProvider(
+    protected val debugPort: Int?,
+) {
+    companion object {
+        const val DEBUG_THREAD_NAME = "NotebookDebugThread"
+    }
+
+    fun provideDebugPort(): Int? = debugPort
+
+    fun provideSuspendedThread(runnable: () -> Unit): Thread {
+        return thread(start = provideDebugPort() != null, name = DEBUG_THREAD_NAME, block = runnable)
+    }
+}
+
+class NotebookDebugIntegrationProvider(
```

Comment:
If it's a `NotebookDebugIntegrationProvider` then why `DebugIntegrationProvider` is missing "Notebook" part in its name?